### PR TITLE
Datagrid: Move 'numberFormat' member to 'e.xlsxCell' in 'onXlsxCellPrepared'

### PR DESCRIPTION
**Pull request by IgnatovDan**

### CODE REVIEW

File: js/client_exporter/excel_creator.js [hunk 1]
```diff
@@ -193,43 +220,40 @@ var ExcelCreator = Class.inherit({
...
+                                cellData.value = this._appendString(modifiedXlsxCell.value);
                                 break;
                             case 'd':
-                                cellData.value = xlsxCell.value;
+                                cellData.value = modifiedXlsxCell.value;
                                 break;
                             case 'n':
-                                let newValue = xlsxCell.value;
+                                let newValue = modifiedXlsxCell.value;
                                 if(typeUtils.isDate(newValue)) {
```
Comment (churkin) [inline js/client_exporter/excel_creator.js]: It is extra check, duplicates https://github.com/DevExpress/DevExtreme/blob/137d19db62516beb71fbdb90f5e1da2ad652ec65/js/client_exporter/excel_creator.js#L134

File: js/client_exporter/excel_creator.js [hunk 2]
```diff
@@ -193,43 +220,40 @@ var ExcelCreator = Class.inherit({
...
-                                    newValue = this._getExcelDateValue(newValue);
-                                    if(!typeUtils.isDefined(newValue)) {
-                                        newValue = xlsxCell.value;
+                                    const xlsxDataValue = this._getExcelDateValue(newValue);
+                                    if(typeUtils.isDefined(xlsxDataValue)) {
```
Comment (churkin) [inline js/client_exporter/excel_creator.js]: `_getExcelDateValue` cant't return `null` or `void 0` in this case

File: js/client_exporter/excel_creator.js [hunk 3]
```diff
@@ -193,43 +220,40 @@ var ExcelCreator = Class.inherit({
...
                 let cellStyleId = dataProvider.getStyleId(rowIndex, cellIndex);
                 if(dataProvider.onXlsxCellPrepared) {
-                    const xlsxCellValue = cellData.sourceValue || cellData.value;
-                    const xlsxCell = {
-                        style: extend(true, {}, that._styleArray[cellStyleId]),
-                        value: xlsxCellValue,
+                    const value = cellData.sourceValue || cellData.value;
+                    const modifiedXlsxCell = this._raiseXlsxCellPrepared({
+                        dataProvider: dataProvider,
+                        value: value,
```
Comment (churkin) [inline js/client_exporter/excel_creator.js:226]:
```javascript
...
dataProvider,
value,
...
```

File: js/client_exporter/excel_creator.js [hunk 4]
```diff
@@ -173,6 +173,33 @@ var ExcelCreator = Class.inherit({
...
+        const numberFormat = style_args.numberFormat;
+        delete style_args.numberFormat;
+
+        const args = {
+            xlsxCell: {
+                style: style_args,
+                value: value,
+                dataType: dataType,
+                numberFormat: numberFormat,
+            },
```
Comment (churkin) [inline js/client_exporter/excel_creator.js:187]:
```javascript
 xlsxCell: {
   style: style_args,
   value,
   dataType,
   numberFormat
 },
```

**Review by churkin — CHANGES_REQUESTED** [review]

**Review by churkin — APPROVED** [review]

File: js/client_exporter/excel_creator.js (hunk 4 above)
Comment (IgnatovDan) [reply js/client_exporter/excel_creator.js:187]: The left parts specify members of an external api. Without this explicit declaration, it is very easy to lost values of that api members when a local variable is renamed. It's better to keep it.

File: js/client_exporter/excel_creator.js (hunk 3 above)
Comment (IgnatovDan) [reply js/client_exporter/excel_creator.js:226]: The left parts specify members of an external api. It's better to keep them.

File: js/client_exporter/excel_creator.js (hunk 1 above)
Comment (IgnatovDan) [reply js/client_exporter/excel_creator.js]: I renamed method and removed unnecessary condition.

File: js/client_exporter/excel_creator.js (hunk 2 above)
Comment (IgnatovDan) [reply js/client_exporter/excel_creator.js]: I renamed method and removed this external condition.

**Pull request merged**